### PR TITLE
Fix inspectTastyFilesInJar in TastyInspector

### LINKED ISSUE #12852: TastyInspector.inspectTastyFilesInJar only inspects top-level tastys
## Compiler version

3.0.0

## Minimized code

See this example project https://github.com/jodersky/tastyjar.

The gist of it is that given a jar file, `TastyInspector.inspectTastyFilesInJar` will only ever read `.tasty` files that are top-level. For instance, assume a jar with content:

```
a.tasty
pkg/b.tasty
```

then only `a.tasty` will be inspected, and `pkg/b.tasty` is silently ignored.

## Expectation

All tasty files in a jar should be inspected.

## Notes

This issue is probably the same underlying cause for https://github.com/lampepfl/dotty/issues/11696

### PR DESCRIPTION
**Pull request by romanowski**
Add integration test

fixes #12852

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Now that we are re-compiling the full std-lib from the jar we hit another issue.
```scala
Test dotty.tools.dotc.BootstrappedStdLibTASYyTest.testFromTastyInJar failed: java.lang.AssertionError: assertion failed: class Unit, took 14.974 sec
```
It looks like `scala/Unit.tasty` is in the blacklist. Maybe there is a bug with the blacklisting logic or the filtering logic.

**Comment by BarkingBad** [comment]
The problem was the paths returned from traversing the Jar file were with prefix root indicator in its own filesystem (e. g. `/scala/Unit.tasty`) and we compared it to relative paths (e. g. `scala/Unit.tasty`). Removing the leading files separator should fix the blacklisting of the primitives and also should work with loading correct class names (the next transformation is replacing all the slashes `/` with the dots `.`, which in fact would create names with leading dot, e. g. `.scala.collection.immutable.List`

File: compiler/src/dotty/tools/io/JarArchive.scala [hunk 1]
```diff
@@ -3,13 +3,16 @@ package dotty.tools.io
 import java.nio.file.{FileSystemAlreadyExistsException, FileSystems}
 
 import scala.jdk.CollectionConverters._
+import java.util.jar.JarFile
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/io/JarArchive.scala:7]: This is not used

**Review by nicolasstucki — APPROVED** [review]
Otherwise LGTM

**Pull request merged**